### ISSUE TITLE
Design and implement mapping of vCPUS and CPU shares to pod requests and limits

### ISSUE THREAD

**Issue by fabiand** [kubevirt/kubevirt#397]
A VM can [have VCPUs](https://libvirt.org/formatdomain.html#elementsCPUAllocation) in addition to that this can be [tuned to add qos constraints](https://libvirt.org/formatdomain.html#elementsCPUTuning).

The vCPU (frontend) related fields which are exposed in the VM.spec.domain part should map to the relevant domxml.
However, there are some discussion points
- [ ] Can we expose cpuset, current and placement as they are very host specific? Could a policy be used instead?
- [ ] What quantity should be exposed on the VM API level (cpu shares, cpu percent, …)

**Comment by sahid** [comment]
I'm questioning myself of the use case behind kubevirt. Basically the aim I do understand is for the industries which have applications not well designed to run on containers. More of that it's probably related to some industries which want take benefit of a cluster managed by k8s but with the performance of hardware (NFV). Based on that I don't think we should try to imitate the behavior of containers with VMs. the concept of cpu shares, cpu percent do not look right. We could reasonable say that the application which are going to run on kubevirt infrastructure want probably full CPUs capability.

**Comment by fabiand** [comment]
It's not about mimicing containers, it's rather about configuring the pod or container to provide the properly configured resource container for the VM. So, the request is far coarse grained than what we can do with VMs. You could also say that the container level resource specification is a rough estimate of what the Vm will require.
And - it's the only way right now, to express the VM requirements to the scheduler.

In other words:
- Let's use the pod/container CPU and memory QoS features, to reserve us a slice
- Let's use libvirt's facilities to perform propper CPU assignment and memory setup (i.e. ballooning)

**Comment by fabiand** [comment]
This was fixed as part of #167

For now we can explicitly request limits/requests for CPU and memory.
